python/pyspark/ml/feature.py

```diff
@@ -392,24 +465,29 @@ class _BucketedRandomProjectionLSHParams:
     .. versionadded:: 3.0.0
     """
 
-    bucketLength = Param(
+    bucketLength: Param[float] = Param(
         Params._dummy(),
         "bucketLength",
         "the length of each hash bucket, " + "a larger bucket lowers the false negative rate.",
         typeConverter=TypeConverters.toFloat,
     )
 
     @since("2.2.0")
-    def getBucketLength(self):
+    def getBucketLength(self) -> float:
         """
         Gets the value of bucketLength or its default value.
         """
-        return self.getOrDefault(self.bucketLength)
+        return self.getOrDefault(self.bucketLength)  # type: ignore[attr-defined]
```

Comment:
This ignore shouldn't be necessary.

Reply:
After checking with mypy: `python/pyspark/ml/feature.py:480: error: "_BucketedRandomProjectionLSHParams" has no attribute "getOrDefault"  [attr-defined]`

Reply:
Updated